irohad/ametsuchi/impl/postgres_query_executor.cpp

```diff
@@ -615,7 +597,7 @@ namespace iroha {
             query_hash_);
       }
 
-      auto ledger_height = block_store_.last_id();
+      auto ledger_height = block_store_.size();
```

Comment:
so you use `size()` to get the last block height, but these values are not guaranteed to be equal as I can see. Why not preserve the `lastId()` method? it could get the difference of cached blocks range (see my comments above).

Reply:
`lastId` method call can be invalid when the block store is empty. It is safe to use `size` in query executor too.

Reply:
okay, then the name can be changed to `topBlockHeight`, while the idea with range difference is the same.